modules/network/subnets.tf

```diff
@@ -5,56 +5,76 @@ resource "oci_core_subnet" "cp" {
   compartment_id             = var.compartment_id
   display_name               = var.label_prefix == "none" ? "control-plane" : "${var.label_prefix}-control-plane"
-  dns_label                  = "cp"
+  dns_label                  = var.label_prefix == "none" ? "cp" : "${var.label_prefix}cp"
   prohibit_public_ip_on_vnic = var.control_plane_type == "private" ? true : false
   route_table_id             = var.control_plane_type == "private" ? var.nat_route_id : var.ig_route_id
   security_list_ids          = [oci_core_security_list.control_plane_seclist.id]
   vcn_id                     = var.vcn_id
+
+  lifecycle {
+    ignore_changes = [dns_label]
+  }
 }
 
 resource "oci_core_subnet" "workers" {
   cidr_block                 = local.workers_subnet
   compartment_id             = var.compartment_id
   display_name               = var.label_prefix == "none" ? "workers" : "${var.label_prefix}-workers"
-  dns_label                  = "workers"
+  dns_label                  = var.label_prefix == "none" ? "workers" : "${var.label_prefix}wo"
   prohibit_public_ip_on_vnic = var.worker_type == "private" ? true : false
   route_table_id             = var.worker_type == "private" ? var.nat_route_id : var.ig_route_id
   vcn_id                     = var.vcn_id
+
+  lifecycle {
+    ignore_changes = [dns_label]
+  }
 }
 
 resource "oci_core_subnet" "pods" {
   cidr_block                 = local.pods_subnet
   compartment_id             = var.compartment_id
   display_name               = var.label_prefix == "none" ? "pods" : "${var.label_prefix}-pods"
-  dns_label                  = "pods"
+  dns_label                  = var.label_prefix == "none" ? "pods" : "${var.label_prefix}po"
   prohibit_public_ip_on_vnic = true
   route_table_id             = var.nat_route_id
   vcn_id                     = var.vcn_id
 
+  lifecycle {
+    ignore_changes = [dns_label]
+  }
+
   count = var.cni_type == "npn" ? 1 : 0
 }
 
 resource "oci_core_subnet" "int_lb" {
   cidr_block                 = local.int_lb_subnet
   compartment_id             = var.compartment_id
   display_name               = var.label_prefix == "none" ? "int_lb" : "${var.label_prefix}-int_lb"
-  dns_label                  = "intlb"
+  dns_label                  = var.label_prefix == "none" ? "intlb" : "${var.label_prefix}ib"
   prohibit_public_ip_on_vnic = true
   route_table_id             = var.nat_route_id
   vcn_id                     = var.vcn_id
 
+  lifecycle {
+    ignore_changes = [dns_label]
+  }
+
   count = var.load_balancers == "internal" || var.load_balancers == "both" ? 1 : 0
 }
 
 resource "oci_core_subnet" "pub_lb" {
   cidr_block                 = local.pub_lb_subnet
   compartment_id             = var.compartment_id
   display_name               = var.label_prefix == "none" ? "pub_lb" : "${var.label_prefix}-pub_lb"
-  dns_label                  = "publb"
+  dns_label                  = var.label_prefix == "none" ? "publb" : "${var.label_prefix}pb"

```

Comment:
Are you sure hyphens are not allowed? https://docs.oracle.com/en-us/iaas/Content/Network/Concepts/dns.htm

Reply:
> VCN and subnet labels: Max 15 alphanumeric characters and must start with a letter. Notice that hyphens and underscores are NOT allowed. The value cannot be changed later.

What do you by labelprefix-shorthand?

Reply:
So instead of `productionwo`, you'll have `production-wo`

Reply:
It's not allowed, I tested it and I got an error. The documentation confirmed it. We can't use dashes and underscores in the dns label.